app/models/manageiq/providers/openstack/inventory/collector/cloud_manager.rb

```diff
@@ -65,7 +65,17 @@ def images
 
   def key_pairs
     return @key_pairs if @key_pairs.any?
-    @key_pairs = compute_service.handled_list(:key_pairs, {}, openstack_admin?)
+    users = User.all
```

Comment:
@andyvesel We might need filter only Users which came from TenantMapping (e.g. with _openstack_user_id_ to not kill refresh in case of many users in CF and let's hope that CU will not have hundreds OpenStack users mapped into CF).

Reply:
added `next if` condition to skip users with `settings.[:openstack_user_id]nil? `

Reply:
Maybe also include admin user